torch/distributed/optim/post_localSGD_optimizer.py

```diff
@@ -65,10 +66,31 @@ def __repr__(self):
         return self.optim.__repr__()
 
     def state_dict(self):
-        return self.optim.state_dict()
+        r"""
+        This is same as :class:`torch.optim.Optimizer` :meth:`state_dict`,
+        but adds an extra entry to record model averager's step to the checkpoint
+        to ensure reload does not cause unnecessary warm up again.
+        """
+        optim_state_dict = self.optim.state_dict()
+        optim_state_dict['step'] = self.averager.step
+        return optim_state_dict
 
     def load_state_dict(self, state_dict):
+        r"""
+        This is same as :class:`torch.optim.Optimizer` :meth:`load_state_dict`,
+        but also restores model averager's step value to the one,
+        saved in provided ``state_dict``.
+
+        If there is no such entry as ``step`` in ``state_dict``,
+        it will raise a warning and initialize model averager's step to 0.
+        """
         self.optim.load_state_dict(state_dict)
+        if 'step' in state_dict:
+            self.averager.step = state_dict['step']
+        else:
+            warnings.warn("Loaded state dict does not contain a step counter for an averager. " +
```

Comment:
nit: I do not think you need a `+` for string concatenation between parentheses.